cluster/cluster_impl/failback_cluster_invoker.go

```diff
@@ -0,0 +1,218 @@
+				retryTask.retries++
+				if retryTask.retries > invoker.maxRetries {
+					perrors.Errorf("Failed retry times exceed threshold (%v), We have to abandon, invocation-> %v",
+						retryTask.retries, retryTask.invocation)
+				} else {
+					invoker.taskList.push(retryTask)
+				}
+			}
+		}
+	}
+}
+
+func (invoker *failbackClusterInvoker) Invoke(invocation protocol.Invocation) protocol.Result {
+
+	invokers := invoker.directory.List(invocation)
+	err := invoker.checkInvokers(invokers, invocation)
+
+	if err != nil {
+		// add retry ticker task
+		perrors.Errorf("Failed to invoke the method %v in the service %v, wait for retry in background. Ignored exception: %v.",
+			invocation.MethodName(), invoker.GetUrl().Service(), err)
+		return &protocol.RPCResult{}
+	}
+	url := invokers[0].GetUrl()
+
+	methodName := invocation.MethodName()
+	//Get the service loadbalance config
+	lb := url.GetParam(constant.LOADBALANCE_KEY, constant.DEFAULT_LOADBALANCE)
+
+	//Get the service method loadbalance config if have
+	if v := url.GetMethodParam(methodName, constant.LOADBALANCE_KEY, ""); v != "" {
+		lb = v
+	}
+	loadbalance := extension.GetLoadbalance(lb)
+
+	invoked := []protocol.Invoker{}
+	var result protocol.Result
+
+	ivk := invoker.doSelect(loadbalance, invocation, invokers, invoked)
+	invoked = append(invoked, ivk)
+	//DO INVOKE
+	result = ivk.Invoke(invocation)
+
+	if result.Error() != nil {
+		// add retry ticker task
+		timerTask := newRetryTimerTask(loadbalance, invocation, invokers, invoker, invoker.maxRetries, 5)
+		invoker.taskList.push(timerTask)
+
+		perrors.Errorf("Failback to invoke the method %v in the service %v, wait for retry in background. Ignored exception: %v.",
+			methodName, url.Service(), result.Error().Error())
+		// ignore
+		return &protocol.RPCResult{}
+	}
+
+	return result
+}
+
+func (invoker *failbackClusterInvoker) Destroy() {
+	invoker.baseClusterInvoker.Destroy()
+
+	// stop ticker
+	invoker.ticker.Stop()
+}
+
+type retryTimerTask struct {
+	loadbalance cluster.LoadBalance
+	invocation  protocol.Invocation
+	invokers    []protocol.Invoker
+	lastInvoker *failbackClusterInvoker
+	retries     int64
+	lastT       time.Time
+}
+
+func newRetryTimerTask(loadbalance cluster.LoadBalance, invocation protocol.Invocation, invokers []protocol.Invoker,
+	lastInvoker *failbackClusterInvoker, retries int64, tick int64) *retryTimerTask {
+	return &retryTimerTask{
+		loadbalance: loadbalance,
+		invocation:  invocation,
+		invokers:    invokers,
+		lastInvoker: lastInvoker,
+		retries:     retries,
+		lastT:       time.Now(),
+	}
+}
+
+type Queue struct {
+	data *list.List
+	sync.Locker
+}
+
+func newQueue() *Queue {
+	q := &Queue{
+		data:   list.New(),
+		Locker: new(sync.Mutex),
+	}
+	return q
+}
+
+func (q *Queue) push(v interface{}) {
+	defer q.Unlock()
+	q.Lock()
+	q.data.PushFront(v)
+}
+
+func (q *Queue) pop() interface{} {
+	defer q.Unlock()
+	q.Lock()
+	iter := q.data.Back()
+	v := iter.Value
+	q.data.Remove(iter)
```

Comment:
if q.data is empty，may nil pointer error.

Reply:
done